tests/lib/nrf_modem_lib/lte_connectivity/src/lte_connectivity_test.c

```diff
@@ -55,72 +69,70 @@ static void network_interface_down_option_set(enum lte_connectivity_if_down_opti
+	}
 
-static void ipv6_address_add(void)
-{
-	struct net_if *net_if = net_if_get_default();
+	/* Temporarily ignore and provide default values to mocked functions */
+	__cmock_lte_lc_func_mode_set_IgnoreAndReturn(0);
+	__cmock_nrf_modem_lib_shutdown_IgnoreAndReturn(0);
 
-	__mock_nrf_modem_at_scanf_ExpectAndReturn(
-			"AT+CGPADDR=0", "+CGPADDR: %*d,\"%46[.:0-9A-F]\",\"%46[:0-9A-F]\"", 2);
-	__mock_nrf_modem_at_scanf_ReturnVarg_string("");
-	__mock_nrf_modem_at_scanf_ReturnVarg_string("2001:8c0:5140:801:1b4d:4ce8:6eed:7b69");
+	/* Lose PDN and CEREG */
+	struct lte_lc_evt cereg_evt = {
+		.type = LTE_LC_EVT_NW_REG_STATUS,
+		.nw_reg_status = LTE_LC_NW_REG_UNKNOWN,
+	};
+	pdn_event_handler_callback(0, PDN_EVENT_DEACTIVATED, 0);
+	lte_lc_event_handler_callback(&cereg_evt);
 
-	pdn_event_handler_callback(0, PDN_EVENT_IPV6_UP, 0);
+	/* Let events propagate */
+	k_sleep(K_MSEC(10));
 
-	TEST_ASSERT_TRUE(net_if_ipv6_get_global_addr(NET_ADDR_PREFERRED, &net_if));
-}
+	/* Remove IP addresses if any */
+	if (net_if_ipv6_get_global_addr(NET_ADDR_PREFERRED, &net_if)) {
+		net_if_config_ipv6_put(net_if);
+	}
 
-void setUp(void)
-{
-	mock_nrf_modem_at_Init();
-}
+	if (net_if_ipv4_get_global_addr(net_if, NET_ADDR_PREFERRED)) {
+		net_if_config_ipv4_put(net_if);
+	}
 
-void tearDown(void)
-{
-	uint8_t option;
-	size_t len;
-	struct net_if *net_if = net_if_get_default();
-	enum lte_lc_func_mode mode_expected = LTE_LC_FUNC_MODE_DEACTIVATE_LTE;
+	/* Reset flags */
+	conn_mgr_if_set_flag(net_if, CONN_MGR_IF_PERSISTENT, false);
 
-	int ret = conn_mgr_if_get_opt(net_if, LTE_CONNECTIVITY_IF_DOWN, (void *)&option, &len);
+	/* Reset options */
+	uint8_t option = LTE_CONNECTIVITY_IF_DOWN_LTE_DISCONNECT;
+	(void)conn_mgr_if_set_opt(net_if, LTE_CONNECTIVITY_IF_DOWN,
+					  (const void *)&option,
+					  sizeof(option));
 
-	TEST_ASSERT_EQUAL(0, ret);
+	/* Take iface down */
+	(void)net_if_down(net_if);
 
-	/* Do teardown depending on the LTE_CONNECTIVITY_IF_DOWN option. */
-	if (net_if_flag_is_set(net_if, NET_IF_UP)) {
-		if (option == LTE_CONNECTIVITY_IF_DOWN_LTE_DISCONNECT) {
-			__cmock_lte_lc_func_mode_set_ExpectAndReturn(mode_expected, 0);
-			TEST_ASSERT_EQUAL(0, net_if_down(net_if));
-		} else if (option == LTE_CONNECTIVITY_IF_DOWN_MODEM_SHUTDOWN) {
-			__cmock_nrf_modem_lib_shutdown_ExpectAndReturn(0);
-			TEST_ASSERT_EQUAL(0, net_if_down(net_if));
-		} else {
-			TEST_ASSERT_TRUE(false);
-		}
-	}
+	/* Let events propagate */
+	k_sleep(K_MSEC(10));
 
-	/* Remove IP addresses if any */
-	if (net_if_ipv6_get_global_addr(NET_ADDR_PREFERRED, &net_if)) {
-		TEST_ASSERT_EQUAL(0, net_if_config_ipv6_put(net_if));
-	}
+	/* Stop ignoring mocked functions*/
+	__cmock_lte_lc_func_mode_set_StopIgnore();
+	__cmock_nrf_modem_lib_shutdown_StopIgnore();
 
-	if (net_if_ipv4_get_global_addr(net_if, NET_ADDR_PREFERRED)) {
-		TEST_ASSERT_EQUAL(0, net_if_config_ipv4_put(net_if));
-	}
+	/* Prepare to track/mock modem_at_scanf calls for the upcoming test */
+	mock_nrf_modem_at_Init();
```

Comment:
I had thought that since `mock_nrf_modem_at` is a custom mock of ours named to look and function similarly to genuine cmock mocks, that a call to `mock_nrf_modem_at_Init` and `_Verify` are required

The github diff kinda butchers this, but both calls were already included before I made any modifications to these tests